src/schema/feeds.ts

```diff
@@ -829,7 +830,14 @@ export const resolvers: IResolvers<any, Context> = traceResolvers({
       (ctx, { source }: SourceFeedArgs, builder, alias) =>
         sourceFeedBuilder(ctx, source, builder, alias),
       feedPageGenerator,
-      applyFeedPaging,
+      (ctx, args, page, builder, alias) => {
+        const qb = builder.orderBy(
+          `CASE WHEN ${alias}.type = '${PostType.Welcome}' THEN 1 ELSE 0 END`,
+          'DESC', // this will be replaced once we introduce the column named `pinned` on v6.
+        );
```

Comment:
The DR has not mentioned a new property to identify whether a post is pinned, understandably since pinning is introduced only at v6. So this boolean identifier will be replaced by that new property.

Though I would like to get some thoughts if anyone wants to introduced `pinned` property now or wait until v6.

Reply:
It's not the right place to put the `orderBy` in. Every feed builder provides a page generator that adds ordering to the query. In this case, we use `feedPageGenerator`. You need to create a new one that will combine pins and creation time

Reply:
Just a note bear in mind that pinning vs welcome post pinning might be different ranking still.
So possibly keep it separate until we start v6 for now :)

Reply:
> Every feed builder provides a page generator that adds ordering to the query.

Yes, that's why I added the `order by` on the paging function of the `sourceFeed` resolver only, so no other feed is affected. But yeah, we can create a new `feedPageGenerator` to make it predictable and reusable.

Reply:
> So possibly keep it separate until we start v6 for now :)

Awesome, thanks for letting me know 🚢

Reply:
@idoshamun just wanted to mention that `order by` commands can be stacked (like it was used here), the final query would contain comma-separated columns for the said command which it will first sort by the order of how it was declared.

So the order by priority would be then:
- pinned
- create date time